server/src/client_api.rs

```diff
@@ -92,7 +92,7 @@ async fn resolve_features(
                 .await
         }
         None => features_cache
-            .get(&cache_key(&edge_token))
+            .get(&cache_key(&validated_token))
```

Comment:
I don't think this line actually matters, as I'm guessing that only the actual token value gets used. So we could remove this if you want to. I just changed it to make it clearer that we're done with the actual edge token at this point.